### ISSUE TITLE
TryGetTimeZoneInfo failed in version 4.0.0

### ISSUE THREAD

**Issue by Patryx** [mattjohnsonpint/TimeZoneConverter#108]
Hi,
Unfortunately in a new version 4.0.0 method **TryGetTimeZoneInfo** works not well.

Take a look:
` var timezones = TimeZoneConverter.TZConvert.KnownWindowsTimeZoneIds.ToList();
 var isFound = TZConvert.TryGetTimeZoneInfo(timeZone, out var timeZone)`

isFound in version 3.5.0 was TRUE, but in version 4.0.0 is here always FALSE.

**Comment by mattjohnsonpint** [comment]
Hi. I'm not sure exactly what you mean, as that code isn't quite complete.

If there's a particular time zone that is not found, please let me know which one.

Also, please let me know which OS and version  you are using, and be sure you have the latest updates.

**Comment by 63Digital** [comment]
I've got a similar issue. I've cloned the repo and you've a couple of failing tests. I *think* the culprit is line [328](https://github.com/mattjohnsonpint/TimeZoneConverter/blob/815c0e28166fa3e60fd373af10336709d1d1a8a5/src/TimeZoneConverter/TZConvert.cs#L328) in TZConvert.cs

I'm on windows so initially systemTimeZones is set to `TimeZoneInfo.GetSystemTimeZones()`

line [330](https://github.com/mattjohnsonpint/TimeZoneConverter/blob/815c0e28166fa3e60fd373af10336709d1d1a8a5/src/TimeZoneConverter/TZConvert.cs#L330) then sets it to the output of `GetSystemTimeZonesLinux()`

I fixed it locally by doing this:
```
if (IsWindows)
{
     systemTimeZones = TimeZoneInfo.GetSystemTimeZones();
}
else
{
    systemTimeZones = GetSystemTimeZonesLinux();
}
```

**Comment by mattjohnsonpint** [comment]
Ahh. I messed that up indeed.  I'll fix later today.  Thanks!

**Comment by saridakis** [comment]
I have the same issue ... you can try the 'Europe/Athens' timezone that is not found ...

**Comment by Patryx** [comment]
@mattjohnsonpint Except this failure, could you tell me if there are any differences in names of timezones between version 3.5.0 and 4.0.0?
To clarify: I have timezone names (timeZoneName) generated by your library v3.5.0 (TZConvert.TryGetTimeZoneInfo(timeZoneName, out var timeZone)) and now I want to use v4.0.0 using the names from v3.5.0. Will it work in all cases?

**Comment by mattjohnsonpint** [comment]
It should still work, yes.  The main difference in recent versions is the data has been updated, and I've dropped support for some older versions of .NET.

I'm AFK much of today, but 'll have an update published later this evening (Pacific time). If this is causing an immediate problem, please use the previous versions until I can publish a fix.

Thanks.

**Comment by mattjohnsonpint** [comment]
Fixed in version 5.0.  Would have just been 4.0.1, but I found and fixed another issue, making a breaking change for the `Known*` collections to be exposed as `IReadOnlyCollection`.
